src/sentry/static/sentry/app/stores/repositoryStore.tsx

```diff
@@ -0,0 +1,88 @@
+import Reflux from 'reflux';
+
+import RepoActions from 'app/actions/repositoryActions';
+import {Repository} from 'app/types';
+
+type RepositoryStoreInterface = {
+  get(
+    orgSlug?: string
+  ): {
+    repos: Array<Repository> | undefined;
+    reposLoading: boolean | undefined;
+    reposError: Error | undefined;
+  };
+
+  orgSlug: string | undefined;
+  lastUpdated: number;
+  repos: Array<Repository> | undefined;
+  reposLoading: boolean | undefined;
+  reposError: Error | undefined;
+
+  loadRepos(orgSlug: string): void;
+  loadReposSuccess(data: Array<Repository>): void;
+  loadReposError(error: Error): void;
+};
+
+const RepositoryStoreConfig: Reflux.StoreDefinition & RepositoryStoreInterface = {
+  orgSlug: undefined,
+  lastUpdated: Date.now(), // withRepositories will refetch if data is older than 30s
+  repos: undefined,
+  reposLoading: undefined,
+  reposError: undefined,
+
+  listenables: RepoActions,
+
+  init() {
+    this.resetRepos();
+  },
+
+  resetRepos() {
+    this.orgSlug = undefined;
+    this.lastUpdated = Date.now();
+    this.repos = undefined;
+    this.reposLoading = undefined;
+    this.reposError = undefined;
+    this.trigger();
+  },
```

Comment:
so the pattern I have used is to store the publicly available properties into a store property e.g. `this.state` - this way you can more easily tell what is or isn't public.

Also we kind of do both patterns, but you can call `trigger` with the updated state which passes it to listeners so you don't have to call `Store.get()`. I would say this pattern is generally better since the data is more stable i.e. if your store listener is async.

But I would consider this nonblocking as I hope this store isn't around long enough.

Reply:
I prefer having trigger() send the store state as well, the HOC components can act as 'lenses' or selectors on the store data.

Reply:
Agreed with both comments here.